### ISSUE TITLE
'ODK Data <> Download Attachments' not working

### ISSUE THREAD

**Issue by mataharimhairi** [hotosm/OpenMapKitServer#83]
KLL cannot download the images attached to the submissions using the 'ODK Data <> Download Attachments' option. Once selected it is stuck on 'Loading data, please wait...'. 

<img width="1280" alt="screen shot 2019-01-25 at 15 52 58" src="https://user-images.githubusercontent.com/8972913/51733804-8d1e6d00-20bd-11e9-861d-502f83a8395f.png">

The images aren't actually showing up on the 'Images' field, but they are definitely there, as the team was able to download 221 .jpg files after the first day in the field. So not sure if it is related to the total download size or because OMK Server is not registering the images as images. 

Potentially linked to this closed issue #54 

![image](https://user-images.githubusercontent.com/8972913/51733829-9f001000-20bd-11e9-95a3-ee4cacea4627.png)

**Comment by willemarcel** [comment]
Hi @mataharimhairi 

I confirmed it's failing. I think the problem is that attachment file is very big, I estimate near 1GB. To make possible to use Basic Authentication when the user downloads the data, I implemented it downloading the file in the browser, saving as a blob in the memory and then making a blob link available to the user.

We need to change it to save the file in the server file system and then generate a link with some random uid in the file name. That way we don't need to make it a blob to handle the authentication system.

**Comment by smit1678** [comment]
@mataharimhairi One thing to check, and I think is probably good practice is the image setting on ODK Collect. Unless there is a need for the original or highest quality photo, it might be helpful to have a lower resolution image uploaded. Not sure the impact on that though.

**Comment by Bhawak** [comment]
@smit1678 yes there is an image quality setting on ODK collect. I have added "Enable image compression" in the setting, which converts the image into 800*800 pixel. The size of the image depends, it's about 250-500 KB if the setting is enabled.

**Comment by mataharimhairi** [comment]
@willemarcel thank you for looking into this so quickly and for providing a solution. would it also be possible for future development, that there would be an option under 'ODK Data' to 'Download Filtered Attachments'? this would help to reduce the download size and allow for a little bit more management of data.

**Comment by willemarcel** [comment]
@mataharimhairi It's a good idea!

**Comment by willemarcel** [comment]
@dakotabenjamin check the last comment from @mataharimhairi . It's what we need to implement. Let me know if you need some support.
